x-pack/plugins/ml/server/routes/schemas/results_service_schema.ts

```diff
@@ -103,3 +103,19 @@ export const getCategorizerStoppedPartitionsSchema = schema.object({
    */
   fieldToBucket: schema.maybe(schema.string()),
 });
+
+export const getDatafeedResultsChartDataSchema = schema.object({
+  /**
+   * Job id to fetch the bucket results for
+   */
+  jobId: schema.string(),
+  timefield: schema.string(),
+  bucketSpan: schema.string(),
+  start: schema.number(),
+  end: schema.number(),
+  desc: schema.maybe(schema.boolean()),
+  /**
+   * Datafeed config for searching source
+   */
+  datafeedConfig: schema.any(),
```

Comment:
the datafeed doesn't need to be passed in if you have the job id, `getDatafeedByJobId` can be used to load the datafeed
https://github.com/elastic/kibana/blob/49a355f96ffbed216c986ad61a3acbee8fdc5c4e/x-pack/plugins/ml/server/models/job_service/datafeeds.ts#L183